### PR TITLE
Bac 73 assign new password

### PR DESCRIPTION
**Pull request by filiusalwin**
BAC-73
New password assign with button in user overview (button not shown at new user screen)
navigates to new page resetPassword.html
used changePassword.js for js
used same value check as well
added get mapping and post mapping in ProfileController

### CODE REVIEW

File: src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java [hunk 1]
```diff
@@ -57,6 +55,33 @@ protected String doChangePassword(Model model, @RequestParam("currentPassword")
...
+        if (user == null || user.isEmpty()) {
+            redirectAttributes.addFlashAttribute("error", "Something went wrong, it looks like this user doesn't exist. You are redirect to the userpage");
+            return "redirect:/user";
+        }
+        user.get().setPassword(passwordEncoder.encode(newPassword));
+        user.get().setPasswordNeedsChange(false);
+        userRepository.save(user.get());
+        model.addAttribute(user.get());
+        redirectAttributes.addFlashAttribute("success", "Password changed successfully.");
+        return "resetPassword";
```
Comment (ViNguyen91) [inline src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java:82]: After change pass successfully, it goes back to the reset pass view, when i accidentally click the submit, it gives 404 error.
Maybe after a success, it goes back to the update modal ?

File: src/main/resources/static/js/changePassword.js [hunk 2]
```diff
@@ -16,4 +16,16 @@ function submitChangePassword(event) {
 
 $(document).ready(function() {
     $("#hintBox").hide();
-});
\ No newline at end of file
+
+});
+
+function submitResetPassword(event) {
```
[the remark this answers is not in the document]
Comment (ViNguyen91) [reply src/main/resources/static/js/changePassword.js:22]: More over, it should also check before submit if the field is empty ->give a prompt that the fields can not be empty/ user has to fill in the fields first

File: src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java [hunk 3]
```diff
@@ -57,6 +55,33 @@ protected String doChangePassword(Model model, @RequestParam("currentPassword")
...
+    @PostMapping("/passwordreset/reset/{userId}")
+    protected String doResetPassword(Model model, @PathVariable("userId") final Integer userId,
+                                      @RequestParam("newPassword") String newPassword, RedirectAttributes redirectAttributes) {
+        Optional<User> user = userRepository.findById(userId);
+        if (user == null || user.isEmpty()) {
+            redirectAttributes.addFlashAttribute("error", "Something went wrong, it looks like this user doesn't exist. You are redirect to the userpage");
+            return "redirect:/user";
+        }
+        user.get().setPassword(passwordEncoder.encode(newPassword));
+        user.get().setPasswordNeedsChange(false);
```
[the remark this answers is not in the document]
Comment (filiusalwin) [reply src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java:78]: Thanks! Set it to true

File: src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java [hunk 4]
```diff
@@ -57,6 +55,33 @@ protected String doChangePassword(Model model, @RequestParam("currentPassword")
         return "changePassword";
     }
 
+    @GetMapping("/passwordreset/{userId}")
+    protected String loadResetPassword(Model model, @PathVariable("userId") final Integer userId, RedirectAttributes redirectAttributes) {
+        Optional<User> user = userRepository.findById(userId);
+        if (user == null || user.isEmpty()) {
+            redirectAttributes.addFlashAttribute("error", "Something went wrong, it looks like this user doesn't exist. You are redirect to the userpage");
```
[the remark this answers is not in the document]
Comment (filiusalwin) [reply src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java:62]: Edited to: "Something went wrong. You are redirect to the user page."

File: src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java [hunk 5]
```diff
@@ -57,6 +55,33 @@ protected String doChangePassword(Model model, @RequestParam("currentPassword")
...
+        model.addAttribute(user.get());
+        return "resetPassword";
+    }
+
+    @PostMapping("/passwordreset/reset/{userId}")
+    protected String doResetPassword(Model model, @PathVariable("userId") final Integer userId,
+                                      @RequestParam("newPassword") String newPassword, RedirectAttributes redirectAttributes) {
+        Optional<User> user = userRepository.findById(userId);
+        if (user == null || user.isEmpty()) {
+            redirectAttributes.addFlashAttribute("error", "Something went wrong, it looks like this user doesn't exist. You are redirect to the userpage");
```
[the remark this answers is not in the document]
Comment (filiusalwin) [reply src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java:74]: "Something went wrong. You are redirect to the userpage." adjusted

File: src/main/resources/static/js/changePassword.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (filiusalwin) [reply src/main/resources/static/js/changePassword.js:22]: Check if field are empty is fixed.
I do  not get the error message when old password is used. It just overwrites it.

File: src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java [hunk 6]
```diff
@@ -57,6 +55,33 @@ protected String doChangePassword(Model model, @RequestParam("currentPassword")
...
+            return "redirect:/user";
+        }
+        user.get().setPassword(passwordEncoder.encode(newPassword));
+        user.get().setPasswordNeedsChange(false);
+        userRepository.save(user.get());
+        model.addAttribute(user.get());
+        redirectAttributes.addFlashAttribute("success", "Password changed successfully.");
```
[the remark this answers is not in the document]
Comment (filiusalwin) [reply src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java:81]: Fixed

File: src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java (hunk 1 above)
Comment (filiusalwin) [reply src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java:82]: Redirected to user page fixed

File: src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (ViNguyen91) [reply src/main/java/nl/miwgroningen/se/ch3/bacchux/controller/ProfileController.java:85]: The message "Password changed successfully" stay at the top of the user page. 
--> Maybe we can show it as a pop-up modal/prompt, and as we ok click, it goes back to the /user? this way we can see the message when the pass is good and we do not have an unmovable message in the /user.

**Pull request merged**